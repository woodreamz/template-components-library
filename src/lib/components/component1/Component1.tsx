import React from 'react';
import styled from 'styled-components';
import { getColor } from '../../utils';

// #region styled-components
const Container = styled.div<{ $color: string }>`
  color: ${({ $color }) => $color};
`;
// #endregion

// #region component
interface Component1Props {
  /** The label to display */
  label: string;
  /** Color of the label */
  color?: string;
}

/**
 * Component1 is a component that displays a label.
 */
const Component1 = ({ label, color = 'green' }: Component1Props) => {
  return <Container $color={getColor(color)}>{label}</Container>;
};
// #endregion

export default React.memo(Component1);
