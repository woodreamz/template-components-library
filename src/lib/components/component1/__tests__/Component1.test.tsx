import { render } from '../../../../tests';
import Component1 from '../Component1';

describe('component1', () => {
  it('render Component1', () => {
    const { container } = render(<Component1 label="label" />);
    expect(container).toMatchSnapshot();
  });
});
