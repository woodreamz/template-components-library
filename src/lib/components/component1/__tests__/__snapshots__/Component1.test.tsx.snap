// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component1 > render Component1 1`] = `
<div>
  <div
    class="sc-bdnyFh jpRFIU"
  >
    label
  </div>
</div>
`;
